### PR TITLE
Upgrade Python dependencies to patch security vulnerabilities

### LINKED ISSUE #400: Upgrade Python dependencies to patch security vulnerabilities
## Need
Update Python `pip` dependencies to patch security vulnerabilities reported by dependabot.

### PR DESCRIPTION
**Pull request by bbusenius**
Fixes #400 

**Changes in this request**
- Major version upgrade of Pillow
- Minor version upgrades of Wagtail and Django
- Patches security vulnerabilities

### CODE REVIEW

**Comment by bbusenius** [comment]
Tagging @bufordrat in this one so that he's aware that I _did_ decide to upgrade Wagtail and Django to patch security vulnerabilities. I was going to hold off but then I realized that they were just minor version upgrades and thus not a big deal. There is a major Wagtail version upgrade available to us but I'll hold off on that one until after we merge digital collections code.

**Review by bufordrat — APPROVED** [review]
Looks good to me!  Runs fine on my DLDC machine w/ the new versions of Django, Wagtail, etc.

**Review by kzadrozny — APPROVED** [review]
Works as expected. No errors occurred when deleting, uploading, or adding images to pages.

**Pull request merged**
